### DISCUSSION TITLE
Containers startup throttling

### DISCUSSION THREAD

**Discussion by chary1112004** [quarkusio/quarkus#34635] [Q&A]
This issue happens for us while deploying quarkus application to kubernetes (eks). It impacts while we upgrade kubernetes while moving pods to new worker nodes and services in many tenants keep crashing as it don't have resource in startup time

When service start, it consumes a lot of cpu ~1000m ~ 1CPU. With this consume then if 1 node contains 4 CPUs then there will be only 4 services could start at the time and other keep crashing and retry until there is free space cpu to reserve.

```
data-service-86995fcbc-vhg4w             1142m        288Mi
```

After service in running time, cpu is quite low ~ 25m

```
data-service-86995fcbc-vhg4w             25m          308Mi
```

Any solution to resolve this issue would be very much appreciated.

Thanks!

[a comment by an automated account is not shown]

**Comment by maxandersen** [comment]
Is this in jvm or native mode?

Have you profiled what parts of the startup that is taking time?

Both core quarkus and User code is involved so without having some profile data it's hard to know where the issue is.

**Reply by geoand** [reply]
In addition to this, it's known that the JVM does a lot of work when the process first boots. 

Have you read [this](https://developers.redhat.com/articles/2022/04/19/best-practices-java-single-core-containers#the_jvm_as_a_dynamic_execution_platform) and [this](https://learn.microsoft.com/en-us/azure/developer/java/containers/overview#determine-how-many-cpu-cores-are-needed) already?

**Comment by chary1112004** [comment]
@maxandersen we are running in jvm mode.

We just did profile in local, not into container part. I will try to remove all code belong to user code and just keep quarkus code and give feedback to you.

Another point that is we have installed top command inside container to verify which process is consuming cpu. Then it is only java that is consuming 200% .
![image](https://github.com/quarkusio/quarkus/assets/76929085/228d9dd1-162f-41eb-9d4f-08872e203dc1)

@geoand thanks for the links. I will check about that.
